api/pull.go

```diff
@@ -260,8 +260,50 @@ func (api *PullAPI) NewFilter(criteria filters.FilterCriteria) (rpc.ID, error) {
 
 // GetFilterLogs returns the logs for the filter with the given id.
 // If the filter could not be found an empty array of logs is returned.
-func (api *PullAPI) GetFilterLogs(ctx context.Context, id rpc.ID) ([]*gethTypes.Log, error) {
-	panic("not implemented")
+func (api *PullAPI) GetFilterLogs(
+	ctx context.Context,
+	id rpc.ID,
+) ([]*gethTypes.Log, error) {
+	api.mux.Lock()
+	defer api.mux.Unlock()
+
+	filter, ok := api.filters[id]
+	if !ok {
+		return nil, errors.Join(
```

Comment:
according to the above description it should return empty array if not found, I suggest adding a test where the filter is not found as well

Reply:
Good point regarding the test case, I will add it right away.
Good catch as well regarding the description. This is actually a wrong description copied from geth, See https://github.com/ethereum/go-ethereum/blob/8d42e115b1cae4f09fd02b71c06ec9c85f22ad4f/eth/filters/api.go#L383-L417 :sweat_smile: I will update the description, the geth behavior is to always return `nil`, instead of an empty array, at least for this specific endpoint.

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/217/commits/d2952b82fef16dea456955c08914e0f3dbed3c8e